### PR TITLE
AEROGEAR-2448 - Create Authentication config implementation

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation
Jira: https://issues.jboss.org/browse/AEROGEAR-2448

## Description
Added the authentication configuration class to the Auth module.  The `AuthenticationConfig` is an immutable object that uses the fluent builder pattern.  Unsure if this is the best approach ¯\_(ツ)_/¯

## Progress
- [x] - `AuthenticationConfig` class
- [x] - docs
- [x] - unit tests

### CODE REVIEW

File: Auth/Config/AuthenticationConfig.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    public class AuthenticationConfig
+    {
+        /// <returns>the redirect uri used during the authentication process</returns>
+        public Uri RedirectUri { get; private set; }
+
+        /// <returns>the OIDC scopes for the auth request. If no scopes are defined, the default 'openid scope will be sent</returns>
+        public string Scopes { get; private set; }
+
+        /// <returns>the minimum time between Json Web Key Set (JWKS) requests in minutes. Default value is 1440 (1 day)</returns>
+        public UInt16 MinTimeBetweenJwksRequests { get; private set; }
```
Comment (wojta) [inline Auth/Config/AuthenticationConfig.cs]: UInt16?  Maybe ushort?

Comment (rachael-oregan) [reply Auth/Config/AuthenticationConfig.cs]: Ya sure.

Just to clarify, it's preferred to use the keyword (ushort) over the system type name (UInt16) ya? So it's more readable

**Comment by rachael-oregan** [comment]
@wojta @ziccardi @aidenkeating mind having a look please?

**Comment by rachael-oregan** [comment]
@aidenkeating ya ok good catch.  I'll update the structure

**Comment by aidenkeating** [comment]
@rachael-oregan Sorry for the quick comment, Visual Studio doesn't seem to have a way to create a nested structure, I think this is good to merge.

@wojta What's the reasoning for the nested structure used in https://github.com/aerogear/aerogear-xamarin-sdk/pull/3 ?

**Comment by wei-lee** [comment]
@aidenkeating I believe it is for namespacing the module. So you will need to do:

```
using AeroGear.Mobile.Core
```

to import the module.

I think we should do the same for the auth module.

Going to create a PR to fix it.

**Pull request merged**